src/v/cloud_storage/remote_partition.cc

```diff
@@ -163,7 +180,7 @@ remote_partition::borrow_result_t remote_partition::borrow_next_segment_reader(
     }
     if (iter == _segments.end()) {
         auto path = manifest.generate_segment_path(*mit);
-        iter = materialize_segment(path, *mit, std::move(segment_unit));
+        iter = get_or_materialize_segment(path, *mit, std::move(segment_unit));
```

Comment:
Maybe we should have a separate call that expects callers to have made the segment check in the same task. Or maybe we can tweak the above `find()` to use `get_or_materialize_segment()`?

Just noting that this is seeking twice for the same offset with no scheduling points in between.

Reply:
I think the offloading logic above should move in `get_or_materialize_segment`, but I chose not to do it in this PR to keep the change less intrusive.